pkg/operator/resource_reconciler.go

```diff
@@ -295,10 +310,17 @@ func (rr *ResourceReconciler) OnDelete(obj interface{}) {
 		return
 	}
 
-	level.Debug(rr.logger).Log("msg", fmt.Sprintf("%s deleted", rr.resourceKind), "key", key)
-	rr.metrics.TriggerByCounter(rr.resourceKind, DeleteEvent).Inc()
+	objMeta, err := meta.Accessor(obj)
+	if err != nil {
+		level.Error(rr.logger).Log("err", fmt.Sprintf("failed to get object meta: %s", err), "key", key)
```

Comment:
It isn't an error since the informer can return a cache.DeletedFinalStateUnknown object.
```suggestion
		return
```